src/daemon/rpmostreed-os.c

```diff
@@ -593,6 +615,29 @@ start_deployment_txn (GDBusMethodInvocation  *invocation,
     return glnx_null_throw (error, "Expected %u fds but received %u",
                             expected_fdn, actual_fdn);
 
+  /* split into two fd lists to make it easier for deploy_transaction_execute */
+  g_autoptr(GUnixFDList) install_local_pkgs = NULL;
+  g_autoptr(GUnixFDList) override_replace_local_pkgs = NULL;
+  if (fd_list)
+    {
+      gint nfds = 0; /* the strange constructions below allow us to avoid dup()s */
+      g_autofree gint *fds = g_unix_fd_list_steal_fds (fd_list, &nfds);
+
+      if (install_local_pkgs_idxs)
+        {
+          gint *new_fds =
```

Comment:
Ahh, right. I misread the doc and thought it took ownership of the array as well. In fact, it's explicitly saying the opposite!